org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/scopes/BuildScopeUnit.scala

```diff
@@ -0,0 +1,82 @@
+package org.scalaide.core.internal.project.scopes
+
+import scala.tools.nsc.Settings
+import org.eclipse.core.resources.IContainer
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.core.runtime.SubMonitor
+import org.eclipse.jdt.core.IJavaModelMarker
+import org.scalaide.core.IScalaProject
+import org.scalaide.core.SdtConstants
+import org.scalaide.core.internal.builder.BuildProblemMarker
+import org.scalaide.core.internal.builder.zinc.SbtAwareBuildManager
+import org.scalaide.core.internal.project.CompileScope
+import sbt.inc.IncOptions
+import org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager
+
+class BuildScopeUnit(val scope: CompileScope, val owningProject: IScalaProject, settings: Settings,
+  val dependentUnitInstances: Seq[BuildScopeUnit] = Seq.empty)
+    extends SbtAwareBuildManager {
+
+  private val delegate =
+    new EclipseSbtBuildManager(owningProject, settings, Some(owningProject.underlying.getFile(".cache-" + scope.name)),
+        addToClasspath, srcOutputs)
+
+  private def isSrcOutPossessor: PartialFunction[(IContainer, IContainer), Boolean] = {
+    case (src, out) if scope.isValidSourcePath(src.getProjectRelativePath) => true
+    case _ => false
+  }
+
+  private def addToClasspath = owningProject.sourceOutputFolders.filterNot(isSrcOutPossessor) map {
+    case (src, out) => out.getLocation
+  }
+
+  private def srcOutputs = owningProject.sourceOutputFolders.filter(isSrcOutPossessor)
+
+  def sources: Seq[IContainer] = srcOutputs.map {
+    case (src, out) => src
+  }
+
+  def clean(implicit monitor: IProgressMonitor) = delegate.clean
+
+  override def build(addedOrUpdated: Set[IFile], removed: Set[IFile], monitor: SubMonitor): Unit = {
+    hasInternalErrors = if (areDependedUnitsBuilt) {
+      def getValidJavaSourcesOfThisScope: Set[IFile] = {
+        val Dot = 1
+        val javaSources = getToCompileFor(owningProject.allSourceFiles
+          .filter { _.getLocation.getFileExtension == SdtConstants.JavaFileExtn.drop(Dot) })
+        javaSources -- removed
+      }
+      def resetJavaMarkers(javaFiles: Set[IFile]): Set[IFile] = {
+        javaFiles.foreach { _.deleteMarkers(IJavaModelMarker.JAVA_MODEL_PROBLEM_MARKER, true, IResource.DEPTH_INFINITE) }
```

Comment:
I am worried about this change. Did you manage to find out why it's needed now? The Java builder manages its own error markers, we should not touch them. It was working before, so is really scary as a change.

I know the problem was a failing test, but that's a manifestation of a deeper error (or a race condition in the test). I think we need to track that down and understand where the issue is.